pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -612,18 +611,47 @@ func (r *hibernationReconciler) checkClusterRunning(cd *hivev1.ClusterDeployment
 		if err := r.updateClusterDeploymentStatus(cd, logger); err != nil {
 			return reconcile.Result{}, err
 		}
-		poolNS, poolName := "", ""
-		if cd.Spec.ClusterPoolRef != nil {
-			poolNS = cd.Spec.ClusterPoolRef.Namespace
-			poolName = cd.Spec.ClusterPoolRef.PoolName
-		}
 		// logging with time since hibernating condition was set to ResumingOrRunning state
-		hivemetrics.MetricClusterReadyTransitionSeconds.WithLabelValues(cd.Labels[constants.VersionMajorMinorPatchLabel],
-			cd.Labels[hivev1.HiveClusterPlatformLabel], poolNS, poolName).Observe(time.Since(hibernatingCondition.LastTransitionTime.Time).Seconds())
+		logCumulativeMetric(hivemetrics.MetricClusterReadyTransitionSeconds, cd,
+			time.Since(hibernatingCondition.LastTransitionTime.Time).Seconds())
+		// Clear entry from currently resuming clusters if exists
+		deleteTransitionMetric(hivemetrics.MetricResumingClustersSeconds, cd)
 	}
 	return reconcile.Result{}, nil
 }
 
+// deleteTransitionMetric has the knowledge of labels needed to clear the metrics for currently transitioning clusters
+// Note: do not use for cumulative metrics - they have different labels and are not designed to be cleared
+func deleteTransitionMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment) {
+	poolNS := "<none>"
+	if cd.Spec.ClusterPoolRef != nil {
+		poolNS = cd.Spec.ClusterPoolRef.Namespace
+	}
+	metric.Delete(map[string]string{
+		"cluster_deployment_namespace": cd.Namespace,
+		"cluster_deployment":           cd.Name,
+		"platform":                     cd.Labels[hivev1.HiveClusterPlatformLabel],
+		"cluster_version":              cd.Labels[constants.VersionMajorMinorPatchLabel],
+		"cluster_pool_namespace":       poolNS,
+	})
+}
+
+// logCumulativeMetric should be used to log cumulative metrics
+func logCumulativeMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment, time float64) {
+	poolNS, poolName := "<none>", "<none>"
+	if cd.Spec.ClusterPoolRef != nil {
+		poolNS = cd.Spec.ClusterPoolRef.Namespace
+		poolName = cd.Spec.ClusterPoolRef.PoolName
+	}
+	if !hivemetrics.ShouldLogDurationMetric(metric, time) {
+		return
+	}
```

Comment:
nit: This block could go at the top. No reason to compute pool name/namespace if we're not gonna use 'em.